src/connection/mod.rs

```diff
@@ -18,11 +19,23 @@ pub fn list() -> Result<Vec<Connection>, String> {
                                 "ListConnections");
     let response = dbus_connect!(message).unwrap();
     let paths: dbus::arg::Array<dbus::Path, _> = response.get1().unwrap();
-    let connections = paths.map(|p| get(p).unwrap()).collect::<Vec<Connection>>();
+    let mut connections: Vec<Connection> = paths.map(|p| get_connection(p).unwrap())
+        .collect();
+    connections.sort_by(|a, b| extract_sort_value(a.clone()).cmp(&extract_sort_value(b.clone())));
```

Comment:
Do you think its possible to use a trait here so that we don't need to explicitly call `extract_sort_value`